numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2044,6 +2044,7 @@ static PyObject *
 array_fromfile(PyObject *NPY_UNUSED(ignored), PyObject *args, PyObject *keywds)
 {
     PyObject *file = NULL, *ret;
+    PyObject *err_ptype = NULL, *err_pvalue = NULL, *err_ptraceback = NULL;
```

Comment:
I'd drop the `p` in the variable names here - PyObjects are always pointers, and never dereferenced in user code.